src/Engine/Scene/SkeletonComponent.hpp

```diff
@@ -0,0 +1,242 @@
+
+    /// \name Component interface
+    /// \{
+
+    void initialize() override {}
+
+    bool canEdit( const Core::Utils::Index& roIdx ) const override;
+
+    Core::Transform getTransform( const Core::Utils::Index& roIdx ) const override;
+
+    void setTransform( const Core::Utils::Index& roIdx, const Core::Transform& transform ) override;
+    /// \}
+
+    /// \name Build from fileData
+    /// \{
+
+    /// Create the skeleton from the given data.
+    void handleSkeletonLoading( const Core::Asset::HandleData* data );
+
+    /// Create the animations from the given data.
+    void handleAnimationLoading( const std::vector<Core::Asset::AnimationData*>& data );
+    /// \}
+
+    /// \name Skeleton-based animation data
+    /// \{
+
+    /// Returns the name of the skeleton.
+    inline const std::string& getSkelName() const { return m_skelName; }
+
+    /// Set the animation skeleton.
+    void setSkeleton( const Core::Animation::Skeleton& skel );
+
+    /// Returns the animation skeleton.
+    inline const Core::Animation::Skeleton* getSkeleton() { return &m_skel; }
+
+    /// Return the number of animations.
+    inline size_t getAnimationCount() const { return m_animations.size(); }
+
+    /// Return the \p i -th animation.
+    inline const Animation& getAnimation( const size_t i ) const { return m_animations[i]; }
+
+    /// Return the \p i -th animation.
+    inline Animation& getAnimation( const size_t i ) { return m_animations[i]; }
+
+    /// Creates a new empty animation from the current pose.
+    Animation& addNewAnimation();
+
+    /// Deletes the \p i-th animation.
+    void removeAnimation( const size_t i );
+
+    /// Set the animation to play.
+    void useAnimation( const size_t i );
+
+    /// Return the index of the animation to play.
+    size_t getAnimationId() const;
+    /// \}
+
+    /// \name Animation Process
+    /// \{
+
+    /// Updates the skeleton pose as the pose corresponding to time \p time.
+    void update( Scalar time );
+
+    /// Returns the current time of animation.
+    Scalar getAnimationTime() const;
+
+    /// Returns the duration of the current animation.
+    Scalar getAnimationDuration() const;
+
+    /// Returns the start and end time of the current animation
+    std::pair<Scalar, Scalar> getAnimationTimeInterval() const;
+
+    /// Set animation speed factor.
+    void setSpeed( const Scalar value );
+
+    /// Return animation speed factor.
+    Scalar getSpeed() const;
+
+    /// Toggle animation auto repeat.
+    void autoRepeat( const bool status );
+
+    /// Return true is animation auto repeat is on, false otherwise.
+    bool isAutoRepeat() const;
+
+    /// Toggle animation ping-pong.
+    void pingPong( const bool status );
+
+    /// Return true is animation ping-pong is on, false otherwise.
+    bool isPingPong() const;
+    /// \}
+
+    /// \name Skeleton display
+    /// \{
+
+    /// Turns xray display on/off for the skeleton bones.
+    void setXray( bool on ) const;
```

Comment:
I guess the Xray concept is somehow outdated and will be removed later ? It does not seems to fit the actual design of Radium.

Reply:
XRay concept is quite outdated but is still managed by the `Renderer` interface. Modifying the way bones are rendered is out of scope of this PR and can be addressed later. I add a todo for this.